nmtwizard/preprocess/tu.py

```diff
@@ -330,10 +364,25 @@ def __init__(
         alignment_log_probs=None,
         source_tokenizer=None,
         target_tokenizer=None,
+        source_context=None,
```

Comment:
I think a constructor argument is not needed. The context can be added with `add_source` after the TU is created, like for fuzzy targets.